cpuburner/cpuburner.go

```
@@ -0,0 +1,209 @@
+	NumBurn int       `json:"num_burn"`
+	TTL     int       `json:"ttl"`
+	ID      uuid.UUID `json:"id,omitempty"`
+}
+
+type cParams struct {
+	Count int
+	TTL   int
+}
+
+type cpuBurnerHandler struct {
+	mutex     sync.Mutex
+	cpuBurner map[uuid.UUID]*CPUBurner
+}
+
+// NewCPUBurnerHandler Handler that returns cpuBurnerHandler with new ID
+func NewCPUBurnerHandler() *cpuBurnerHandler {
+
+	return &cpuBurnerHandler{
+		cpuBurner: make(map[uuid.UUID]*CPUBurner),
+		mutex:     sync.Mutex{},
+	}
+}
+
+// RemoveStoppedJobs stopped jobs cleanner
+func (c *cpuBurnerHandler) RemoveStoppedJobs() {
+	go removeJobs(c)
+}
+
+func removeJobs(c *cpuBurnerHandler) {
+	for {
+		time.Sleep(1 * time.Second)
+		for _, cs := range c.cpuBurner {
+			if !cs.Running {
+				c.mutex.Lock()
+				delete(c.cpuBurner, cs.ID)
+				c.mutex.Unlock()
+			}
+		}
+	}
+}
+
+// CPUBurnerHandler HTTP handler that returns cpuBurner state
+func (c *cpuBurnerHandler) CPUBurnerHandler(w http.ResponseWriter, r *http.Request) {
+	defer c.mutex.Unlock()
+	c.mutex.Lock()
+	idRaw, found := mux.Vars(r)["id"]
+	if !found {
+		w.WriteHeader(http.StatusNotFound)
+		w.Header().Set("Content-Type", "application/json")
+		data, err := util.GetHTTPError("ID not found")
+		if err != nil {
+			log.Fatalf("Cannot serialize error %s", err.Error())
+		}
+		w.Write(data)
+		return
+	}
+	id, err := uuid.FromString(idRaw)
+	if err != nil {
+		fmt.Printf("Something went wrong: %s", err)
+		return
+	}
+	if cs, ok := c.cpuBurner[id]; ok {
+		data, err := json.Marshal(cs)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			io.WriteString(w, "Failed to serialize output!")
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(data)
+		return
+	}
+	w.WriteHeader(http.StatusNotFound)
+	w.Header().Set("Content-Type", "application/json")
+	data, err := util.GetHTTPError("ID not found")
+	if err != nil {
+		log.Fatalf("Cannot serialize error %s", err.Error())
+	}
```

usually not a great idea to fatal error. It's probably better to return an internal error.